### ISSUE TITLE
Authentication failing

### ISSUE THREAD

**Issue by aowzone** [AngellusMortis/sxm-player#11]
* sxm-player version: sxm-0.2.8-py3-
* Python version: 3.8
* Operating System: Windows 10 Pro

### Description
Trying to generate local m3u8 link for streaming to another room.

### What I Did

ran "sxm" and authenticated

got error below.  Seems similar to this:  https://github.com/andrew0/SiriusXM/issues/34

```
ERROR:C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\sxm\client.py:Unable to authenticate because login failed
Traceback (most recent call last):
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\tenacity\_asyncio.py", line 46, in call
    result = yield from fn(*args, **kwargs)
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\sxm\client.py", line 305, in authenticate
    raise AuthenticationError("Reset session")
sxm.client.AuthenticationError: Reset session

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "c:\users\someuser\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\scripts\sxm.exe\__main__.py", line 7, in <module>
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\sxm\__main__.py", line 14, in start
    typer.run(main)
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\typer\main.py", line 864, in run
    app()
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\typer\main.py", line 214, in __call__
    return get_command(self)(*args, **kwargs)
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\click\core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\click\core.py", line 782, in main
    rv = self.invoke(ctx)
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\click\core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\click\core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\typer\main.py", line 500, in wrapper
    return callback(**use_params)  # type: ignore
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\sxm\cli.py", line 88, in main
    run_http_server(sxm, port, ip=host, precache=precache)
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\sxm\http.py", line 184, in run_http_server
    if not sxm.authenticate():
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\sxm\client.py", line 938, in authenticate
    return make_sync(self.async_client.authenticate)()
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\make_it_sync\func_wrapper.py", line 63, in wrapped_call
    return _sync_version_of_function(fn, *args, **kwargs)
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\make_it_sync\func_wrapper.py", line 14, in _sync_version_of_function
    return loop.run_until_complete(r)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\asyncio\base_events.py", line 616, in run_until_complete
    return future.result()
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\tenacity\_asyncio.py", line 43, in call
    do = self.iter(retry_state=retry_state)
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\tenacity\__init__.py", line 332, in iter
    six.raise_from(retry_exc, fut.exception())
  File "<string>", line 3, in raise_from
tenacity.RetryError: RetryError[<Future at 0x282c3929b20 state=finished raised AuthenticationError>]
Exception ignored in: <function SXMClientAsync.__del__ at 0x00000282C33F2280>
Traceback (most recent call last):
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\sxm\client.py", line 158, in __del__
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\make_it_sync\func_wrapper.py", line 63, in wrapped_call
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\make_it_sync\func_wrapper.py", line 14, in _sync_version_of_function
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\asyncio\base_events.py", line 616, in run_until_complete
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\sxm\client.py", line 504, in close_session
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\httpx\_client.py", line 1946, in aclose
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\httpx\_transports\default.py", line 365, in aclose
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\httpcore\_async\connection_pool.py", line 303, in aclose
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\httpcore\_synchronization.py", line 15, in __aenter__
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\anyio\_core\_synchronization.py", line 117, in acquire
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\anyio\lowlevel.py", line 42, in checkpoint_if_cancelled
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\anyio\_core\_eventloop.py", line 134, in get_asynclib
  File "C:\Users\someuser\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\sniffio\_impl.py", line 81, in current_async_library
sniffio._impl.AsyncLibraryNotFoundError: unknown async library, or not in async context
```

**Comment by parkerlreed** [comment]
Yeah looks like it's bbroken. Here's sxm-player with verbose

```
2022-03-19 18:55:35 wolfcola sxm_player[6670] DEBUG Starting worker: status_check
2022-03-19 18:55:35 wolfcola sxm_player.status_check[6674] DEBUG Entering init_signals
2022-03-19 18:55:35 wolfcola sxm_player[6670] DEBUG Startup Event: status_check got True
2022-03-19 18:55:35 wolfcola sxm_player[6670] INFO Attempting to connect SXM Client (next in 10 seconds)
2022-03-19 18:55:35 wolfcola sxm_player[6670] DEBUG Starting worker: sxm
2022-03-19 18:55:36 wolfcola asyncio[6675] DEBUG Using selector: EpollSelector
2022-03-19 18:55:36 wolfcola /usr/lib/python3.10/site-packages/sxm/client.py[6675] DEBUG Logging in as my@email.com...
2022-03-19 18:55:37 wolfcola httpx._client[6675] DEBUG HTTP Request: POST https://player.siriusxm.com/rest/v2/experience/modules/modify/authentication "HTTP/1.1 200 OK"
2022-03-19 18:55:37 wolfcola /usr/lib/python3.10/site-packages/sxm/client.py[6675] ERROR Unable to authenticate because login failed
```

**Comment by mackid1993** [comment]
Having the same problem here, I thought it was just me. Just to add, after trying a few times I actually found myself IP blocked from https://player.siriusxm.com. I had to restart my router and it grabbed a new IP from my ISP to access the website again.

**Comment by carseneau** [comment]
Can confirm same issue here as @mackid1993  with being ip blocked and having login failing for just under a week now

**Comment by aowzone** [comment]
No IP block here. Web player is still all good.

**Comment by DCH3416** [comment]
Also observing.

**Comment by jmperagine** [comment]
I had it working on a Raspberry Pi running Moode Audio, quit working a few days ago. I can login with browser and Grace devices so they're not blocking my IP.

**Comment by highcountrypc** [comment]
no IP block here. been having the same issues for about a week now. hoping someone can come up with a similar fix as the last cookie change/fix.

**Comment by Zendog** [comment]
Looks like it is working again now ...... Yes?

python3 /Users/xyz/test/old_sxm.py userhere pwdhere
*************************************************************************
The SXMAUTHNEW value is below
1.Now ==> True
*************************************************************************
*************************************************************************
The AWSALB value is below
2.Now ==> True
*************************************************************************
channelGuid
channelId
name
shortName
streamingName
sortOrder
shortDescription
mediumDescription
url
clientBufferDuration
inactivityTimeout
isAvailable
isMature
isMySxm
isFavorite
isPersonalized
spanishContent
isBizMature
channelNumber
siriusChannelNumber
xmChannelNumber
xmServiceId
images
geoRestrictions
categories
disableRecommendations
isPlayByPlay
aodShowCount
satOnly
subscribed
ipOnly
channelContentType
darkBackgroundColor
lightBackgroundColor
liveVideoEligible
fastEnabled
esid
adRepEnabled
adBreakDuration
pittsburgh steelers 9170 102fb848-1671-f633-f0c3-c3cbc9775af0
ID                  | Num | Name                    | Guid                                 
siriushits1         | 2   | SiriusXM Hits 1         | 194adbca-34d6-cb94-b153-3488ee563308 
9492                | 3   | Pandora Now             | 02394c8d-7afe-6ba2-3791-fce63b1e4d85 
9544                | 4   | TikTok Radio            | 6e59f0fe-6fcc-ef8f-0ba7-72eb77385eb0 
thepulse            | 5   | The Pulse               | 9e8d6f72-0b59-85cf-a222-b18d38acdc0f 
coffeehouse         | 6   | The Coffee House        | 41f8b174-fb06-1707-aaa7-e74e109c2ee9 
totally70s          | 7   | 70s on 7                | 95d4d8ef-55e4-8337-78f1-2fda944991d8 
big80s              | 8   | 80s on 8                | 2ea07147-a720-ed0c-d4ce-d7bddd1640d3 
8206                | 9   | 90s on 9                | c9256031-1123-f35f-4d49-1ba4d5379258 
8208                | 10  | Pop2K                   | 40c3d189-9611-e59f-5808-7847e294642e 
9556                | 11  | The 10s Spot            | 038e9a9a-4878-7561-521b-5d432a0798a0 
9450                | 12  | PopRocks                | 44f9129f-579a-3d23-218f-3c3518036fc6 
9406                | 13  | Pitbull's Globalization | 5d71ae5d-5c6e-0430-6fc5-d9ca0c1c34c0 
8241                | 14  | KIIS-Los Angeles        | 9d5ce994-5766-2444-8782-186bb92ee066 
9557                | 15  | Mosaic                  | d0759ae0-eeef-48e4-c4b9-526217aac33d 
starlite            | 16  | The Blend               | bda81f03-f231-d4db-56ad-40cb132c5663 
thebridge           | 17  | The Bridge              | 1898923d-071d-0abd-1a62-ed0d2702d4f1 
9446                | 18  | The Beatles Channel     | 20801778-3df2-d607-eab5-e03de0ca7815 
9523                | 19  | Bob Marley's Tuff Gong  | 881c9b1f-e7fd-4637-b87a-0dd06f1831a5 
estreetradio        | 20  | E Street Radio          | 727fa612-6e53-458e-720d-b662dc974e5d 
undergroundgarage   | 21  | Underground Garage      | 3b83dc80-5987-6898-e98d-a98012f700c1 
8370                | 22  | Pearl Jam Radio         | cdc9bee8-9317-37fa-ec8e-464096670c60 
gratefuldead        | 23  | Grateful Dead           | 067801cb-bb3f-1707-dd21-d77e06bb27c0 
radiomargaritaville | 24  | Radio Margaritaville    | c636d008-7cb0-99eb-7fb7-7c489f862a26 
classicrewind       | 25  | Classic Rewind          | 7a642de7-c33f-a628-efb2-3d94a829d17b 
classicvinyl        | 26  | Classic Vinyl           | 5ad8659a-414a-9e26-b973-f5a229d788dd 
thevault            | 27  | Deep Tracks             | e3041d19-daa5-6517-8c73-41976582d1f9 
thespectrum         | 28  | The Spectrum            | 804a5411-b234-1188-27da-f01197103886 
9139                | 29  | Phish Radio             | 0b5b5f60-dfcf-eba9-1dc6-ec3c909f4541 
9506                | 30  | Dave Matthews Band Radio| 69e5185d-d218-8fbc-082f-6ea5a16ca4db 
9407                | 31  | Tom Petty Radio         | 60b0ea2e-dec7-1d4a-da2a-6de09f273fe7 
9507                | 32  | U2 X-Radio              | a4b7a4f7-af96-3ae6-53df-f78ff9ac9aab 
firstwave           | 33  | 1st Wave                | 65f04311-3581-256c-97b9-279838d6ff5e 
90salternative      | 34  | Lithium                 | 32c4747f-8739-d578-62d8-08a92d518445

**Comment by aowzone** [comment]
Seemingly, yes!

**Comment by jmperagine** [comment]
It's working! This makes me extremely happy.

**Comment by mackid1993** [comment]
Working here too, this is great!

**Comment by parkerlreed** [comment]
Working here too

```
2022-03-23 02:23:44 wolfcola sxm_player[22365] INFO Attempting to connect SXM Client (next in 10 seconds)
2022-03-23 02:23:46 wolfcola sxm_player.sxm[22370] INFO sxm has started on http://127.0.0.1:9999
2022-03-23 02:23:46 wolfcola sxm_player[22365] INFO SXM Client started. 114 channels available
```

**Comment by ocdevo** [comment]
I'm wondering if there is some sort of slow replication happening on SiriusXM's side. I just created a new account (Been leveraging the Free trial without an account in the car for a month now. I have not been able to get authenticated with the player. I'm running in Docker and this is what I am seeing:
```
sxm-player_1  | Running HLS proxy...
sxm-player_1  | TERM environment variable not set.
sxm-player_1  | 2022-03-29 20:03:01 6b3d4040a014 sxm_player[8] INFO Attempting to connect SXM Client (next in 10 seconds)
sxm-player_1  | 2022-03-29 20:03:03 6b3d4040a014 /usr/local/lib/python3.9/site-packages/sxm/client.py[12] ERROR Unable to authenticate because login failed
sxm-player_1  | 2022-03-29 20:03:11 6b3d4040a014 /usr/local/lib/python3.9/site-packages/sxm/client.py[12] ERROR An Exception occurred when trying to perform the POST request!
sxm-player_1  | 2022-03-29 20:03:11 6b3d4040a014 /usr/local/lib/python3.9/site-packages/sxm/client.py[12] ERROR Params: {'moduleList': {'modules': [{'moduleRequest': {'resultTemplate': 'web', 'deviceInfo': {'osVersion': 'Windows', 'platform': 'Web', 'sxmAppVersion': '5.36.514', 'browser': 'Firefox', 'browserVersion': '89.0', 'appRegion': 'US', 'deviceModel': 'EverestWebClient', 'clientDeviceId': 'null', 'player': 'html5', 'clientDeviceType': 'web'}, 'standardAuth': {'username': 'xxxxxx@xxxxx.com', 'password': 'xxxxxxxxxxxxxxxxxxxxxx'}}}]}}
sxm-player_1  | 2022-03-29 20:03:11 6b3d4040a014 /usr/local/lib/python3.9/site-packages/sxm/client.py[12] ERROR Method: POST
sxm-player_1  | 2022-03-29 20:03:11 6b3d4040a014 /usr/local/lib/python3.9/site-packages/sxm/client.py[12] ERROR Request: <Request('POST', 'https://player.siriusxm.com/rest/v2/experience/modules/modify/authentication')>
sxm-player_1  | Process Process-2:
sxm-player_1  | Traceback (most recent call last):
sxm-player_1  |   File "/usr/local/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
sxm-player_1  |     self.run()
sxm-player_1  |   File "/usr/local/lib/python3.9/multiprocessing/process.py", line 108, in run
sxm-player_1  |     self._target(*self._args, **self._kwargs)
sxm-player_1  |   File "/usr/local/lib/python3.9/site-packages/sxm_player/runner.py", line 50, in worker_wrapper
sxm-player_1  |     worker = worker_class(*args, **kwargs)
sxm-player_1  |   File "/usr/local/lib/python3.9/site-packages/sxm_player/workers/server.py", line 50, in __init__
sxm-player_1  |     self.sxm.authenticate()
sxm-player_1  |   File "/usr/local/lib/python3.9/site-packages/sxm/client.py", line 938, in authenticate
sxm-player_1  |     return make_sync(self.async_client.authenticate)()
sxm-player_1  |   File "/usr/local/lib/python3.9/site-packages/make_it_sync/func_wrapper.py", line 63, in wrapped_call
sxm-player_1  |     return _sync_version_of_function(fn, *args, **kwargs)
sxm-player_1  |   File "/usr/local/lib/python3.9/site-packages/make_it_sync/func_wrapper.py", line 14, in _sync_version_of_function
sxm-player_1  |     return loop.run_until_complete(r)
sxm-player_1  |   File "/usr/local/lib/python3.9/asyncio/base_events.py", line 629, in run_until_complete
sxm-player_1  |     self.run_forever()
sxm-player_1  |   File "/usr/local/lib/python3.9/asyncio/base_events.py", line 596, in run_forever
sxm-player_1  |     self._run_once()
sxm-player_1  |   File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1854, in _run_once
sxm-player_1  |     event_list = self._selector.select(timeout)
sxm-player_1  |   File "/usr/local/lib/python3.9/selectors.py", line 469, in select
sxm-player_1  |     fd_event_list = self._selector.poll(timeout, max_ev)
sxm-player_1  |   File "/usr/local/lib/python3.9/site-packages/sxm_player/signals.py", line 29, in default_signal_handler
sxm-player_1  |     raise exception_class()
sxm-player_1  | sxm_player.signals.TerminateInterrupt
sxm-player_1  | 2022-03-29 20:03:11 6b3d4040a014 sxm_player[8] INFO Terminated sxm after 3 attempt(s)
sxm-player_1  | 2022-03-29 20:03:11 6b3d4040a014 sxm_player[8] ERROR Exception: Process sxm failed to startup after 10.0 seconds
```
My Experience in browsers have been spotty, usually after trying with the player I get a message that my account has been temporarily disabled for the next (insert time here) Times have varied from 3 to 20 minutes. Then it has also randomly just said it did not recognize my username and password. When watching from the browser I get this response to the authentication call:
```
{
    "ModuleListResponse": {
        "messages": [
            {
                "message": "Account Locked",
                "code": 102
            }
        ],
        "status": 0,
        "moduleList": {
            "modules": [
                {
                    "moduleArea": "UserFlow",
                    "moduleType": "StandardAuth",
                    "updateFrequency": 0,
                    "wallClockRenderTime": "2022-03-29T20:14:20.060+0000",
                    "moduleResponse": {
                        "authenticationData": {
                            "remainingLockoutMinutes": 3,
                            "remainingLockoutSeconds": 53
                        }
                    }
                }
            ]
        }
    }
}
```
I do have two IP Addresses (They are load balanced) so its possible my server with the docker container went out on a different IP Address than my local machine where I had setup my SiriusXM Account. One of my browsers did pop up asking if I wanted to continue listening here instead of the other location.

I am going to let things settle down for a few hours then try again and hopefully get a successful connection like others here. One thing we may be able to do is look at the response from the Authentication call for the sample failed call then try again after the lockout time specified. The Authentication call did come back with a 200 status code. I'm new to coding so fixing this may not be in my wheel house. just wanted to share my experience here for the group.

**Comment by jmperagine** [comment]
It's broken again, worked on March 29, now getting "connection refused" error on March 30.

**Comment by Zendog** [comment]
The original (authentication issues) that were initially reported in this thread continue to work in my testing. Not seeing issues in my environment. The script function seems to login in fine and is authenticating the session. Once logged in, the channels are available and serving up fine.

**Comment by ocdevo** [comment]
Feels like SiriusXM may have some weird login protection logic in place. I was going to give up on this project and just play directly from the website on a browser on my server. After signing in via the browser and playing some music for a few hours, I tried SXM-Player one last time before deleting everything and it successfully logged in. I am currently playing from my docker container without any issues.

It may be that Account changes take ~24 hours to replicate across their various systems. Also as I mentioned in my previous post the web site will show you a time frame if your account has gotten locked out. Testing login with the SXM-Player is a very easy way to trigger this lockout. Continued lockouts can cause your IP to get blocked by a firewall on the Sirius side. (I got presented a block page from a firewall appliance)

If someone has the capability (I'm still new to python / coding) we could add some additional logic in the login process to look for these account lockouts. I'm having issues with getting the application to run locally (I have it running in a docker container) but that is a separate issue. At the very least I will add some detail to the documentation

**Comment by jmperagine** [comment]
oh boy, and just like that it's working

**Comment by Zendog** [comment]
Anyone else having issues again? It looks like they may have recently messed around with the AWS cookies again. This may have impacted the return value string for the authentication processing.

**Comment by Zendog** [comment]
FYI .... If you hit this .... I was able to kickstart the authentication again this afternoon.

Was getting "Unable to authenticate because login failed" when trying to connect. 

Could login successfully using https://player.siriusxm.com/ so credentials were correct. 

Examining the cookies within the browser seem to indicate that SXM has now changed "SXMAUTHNEW" to "SXMDATA". 

Updated code and was able to authenticate again and list/play channels.

find the following line: 

return 'SXMAUTHNEW' in self.session.cookies

and change it to 'SXMDATA' and you should be all set!

**Comment by threedotsonedash** [comment]
Solved it for me, thanks!

**Comment by asingletary** [comment]
Thank you @Zendog this works.

**Comment by CHJ85** [comment]
Thank you @Zendog

**Comment by jmperagine** [comment]
ugh, Sirius blocking my IP, can't even login with browser

update, can't even log into the website to check my account, they are experiencing "technical difficulties" hmm...

However, the android app works

Ha! I did nothing and 24 hours later it works. I am so confused.

**Comment by Resinchem** [comment]
Just an update as of Nov 2023.  I had been running the server for multiple months with no issue.  Suddenly the app quit working, but could still log in via browser and stream.  However in the process of trying to fix the streaming server (docker-compose, multiple restarts,  reboot of the container, etc.), the web site (https://player.siriusxm.com/) suddenly stopped and I received an "Access Denied - Error 16 - The request was blocked  by our security services", followed by my IP address.  This block was still in place after 24 hours and none of my local device could stream (e.g. Roku, Google speakers, etc.). But I could still stream via cellular on mobile and car was fine, so it wasn't my account.   Called SiriusXM (you can already imagine how that went) and eventually after escalation, they ended up blaming the ISP (Comcast) for blocking SiriusXM... yeah, right!  It was THEIR message that said I had been blocked, yet they claimed that they do not block IP addresses.

I figure that a new IP address will resolve the situation, but with Comcast getting a new IP generally involves disconnecting the modem for an extended period.  Just wanted to pass this along and see if anyone else has recently had their IP block from all streaming sources via IP.
